packages/core/src/new-api/internal/module-builder.ts

```diff
@@ -364,6 +367,52 @@ export class IgnitionModuleBuilderImplementation<
     return future;
   }
 
+  public readEventArgument(
+    futureToReadFrom:
+      | NamedContractDeploymentFuture<string>
+      | ArtifactContractDeploymentFuture
+      | NamedContractCallFuture<string, string>,
+    eventName: string,
+    argumentName: string,
+    options: ReadEventArgumentOptions = {}
+  ): ReadEventArgumentFuture {
+    const eventIndex = options.eventIndex ?? 0;
+
+    const futureToReadFromsContract =
```

Comment:
for readability and clarity, can we change this variable name to `contractToReadFrom` or something similar?